### PR TITLE
SignIn Post Request Completed

### PR DESCRIPTION
**Pull request by ArnavParekhji**

### CODE REVIEW

File: app/src/main/java/io/github/djunicode/canteenapp/BaseActivity.java [hunk 1]
```diff
@@ -1,21 +1,72 @@
...
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        toast = this.getIntent().getStringExtra("toastMessage");
+        if (toast != null) {
+            Toast.makeText(getApplicationContext(), toast, Toast.LENGTH_SHORT).show();
+        }
+
+        OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
```
Comment (Ruturaj123) [inline app/src/main/java/io/github/djunicode/canteenapp/BaseActivity.java:40]: Why exactly do we need OkHttpClient? Is it something that cannot be done using Retrofit?

File: app/src/main/java/io/github/djunicode/canteenapp/BaseActivity.java [hunk 2]
```diff
@@ -1,21 +1,72 @@
...
+                    Intent intent = new Intent(getApplicationContext(), LoginActivity.class).putExtra(
+                            "toastMessage", "Security Credentials Invalid"
+                    );
+                    startActivity(intent);
+                    return response;
+                }
+                return response;
+            }
+        });
+        customRetrofit = new Retrofit.Builder()
```
Comment (Ruturaj123) [inline app/src/main/java/io/github/djunicode/canteenapp/BaseActivity.java:61]: Please create a separate Java file for defining the retrofit instance as you will be needing it in multiple places

File: app/src/main/java/io/github/djunicode/canteenapp/BaseActivity.java (hunk 1 above)
Comment (ArnavParekhji) [reply app/src/main/java/io/github/djunicode/canteenapp/BaseActivity.java:40]: I added that so i can increase the timeout, incase of slow connections the request will not get completed

File: app/src/main/java/io/github/djunicode/canteenapp/BaseActivity.java (hunk 2 above)
Comment (ArnavParekhji) [reply app/src/main/java/io/github/djunicode/canteenapp/BaseActivity.java:61]: We can extend base activity in all other activities so the same instance can be used without putting in a separate class

File: app/src/main/java/io/github/djunicode/canteenapp/BaseActivity.java (hunk 1 above)
Comment (Ruturaj123) [reply app/src/main/java/io/github/djunicode/canteenapp/BaseActivity.java:40]: Okay :+1:

**Comment by Ruturaj123** [comment]
Also if a PR is still under development then just put WIP at the start of the PR name to avoid merging of incomplete PRs.

**Pull request merged**